pkg/devicetemplateconverter/devicetemplateconverter.go

```diff
@@ -45,12 +46,24 @@ var errNotFound = errors.DefineNotFound("converter", "converter `{id}` not found
 
 // New returns a new *DeviceTemplateConverter.
 func New(c *component.Component, conf *Config) (*DeviceTemplateConverter, error) {
+	// Overwrites the TTSCSV converter in order to include a profile fetcher.
+	tmplFetcher, err := profilefetcher.NewTemplateFetcher(c)
+	if err != nil {
+		return nil, err
+	}
+	devicetemplates.RegisterConverter(
+		devicetemplates.TTSCSV,
+		devicetemplates.NewCSV(
+			profilefetcher.NewFetcherByVendorIDs(tmplFetcher),
+			profilefetcher.NewFetcherByVersionIDs(tmplFetcher),
+		),
+	)
```

Comment:
What happens now is that a perceived singleton (the TTSCSV converter) is replaced by one that uses the last returned `profilefetcher.NewTemplateFetcher(c)`.

Instead, can we please put the fetcher in the context so that the singleton can extract it?

Reply:
A good idea, I took this a step further and changed so that there is no longer the `NewCSV` function and the CSV converter is declared how it previously was but now validates for the fetcher and if its `nil` the procedure is skipped. 
Updated the tests to accommodate for the changes as well, let me know what you think.